moped-editor/src/views/projects/projectView/ProjectFiles.js

```diff
@@ -194,37 +206,76 @@ const ProjectFiles = props => {
 
   return (
     <CardContent>
-      <Grid container spacing={2}>
-        <Grid item xs={12} md={12}>
-          <Grid container>
-            <Grid sm={12} md={6}>
-              <h2 className={classes.title}>Files</h2>
-            </Grid>
-            <Grid sm={12} md={6}>
-              <Button
-                variant="contained"
-                color="primary"
-                className={classes.uploadFileButton}
-                startIcon={<Icon>backup</Icon>}
-                onClick={handleClickUploadFile}
-              >
-                Upload File
-              </Button>
-            </Grid>
-          </Grid>
-        </Grid>
-      </Grid>
       <ApolloErrorHandler errors={error}>
         <MaterialTable
           columns={columns}
           data={data?.moped_project_files ?? null}
-          title={null}
+          title={
+            <Typography variant="h2" color="primary">
+              Files
+            </Typography>
+          }
+          // Action component customized as described in this gh-issue:
+          // https://github.com/mbrn/material-table/issues/2133
+          components={{
+            EditRow: props => (
+              <MTableEditRow
+                {...props}
+                onKeyDown={e => {
+                  if (e.keyCode === 13) {
+                    // Bypass default MaterialTable behavior of submitting the entire form when a user hits enter
+                    // See https://github.com/mbrn/material-table/pull/2008#issuecomment-662529834
+                  }
+                }}
+              />
+            ),
+            Action: props => {
+              // If isn't the add action
+              if (
+                typeof props.action === typeof Function ||
+                props.action.tooltip !== "Add"
+              ) {
+                return <MTableAction {...props} />;
+              } else {
+                return (
+                  <Button
+                    variant="contained"
+                    color="primary"
+                    size="large"
+                    className={classes.uploadFileButton}
+                    startIcon={<AddCircleIcon />}
+                    ref={addActionRef}
+                    onClick={handleClickUploadFile}
+                  >
+                    Add file
+                  </Button>
+                );
+              }
+            },
+          }}
           icons={{ Delete: ClearIcon }}
           options={{
-            search: true,
+            ...(data.moped_project_files.length < 26 && {
```

Comment:
Took me a second to understand why this was 26. In general, random hardcoded values should be avoided when possible. I wonder if we could use a variable that is set somewhere else as the paging default. If not, even defining a constant as...

```js
const PAGING_DEFAULT_COUNT = 25;

...
# inside options prop
...(data.moped_project_files.length < PAGING_DEFAULT_COUNT + 1 && {
```
maybe thats easier to read?

Reply:
Thanks for pointing this out! Created a tables file to store this value so it can be changed globally and placed it in a new constants folder where we can store other values like this.

Reply:
This concept in software development is called "magic numbers" 
https://refactoring.guru/replace-magic-number-with-symbolic-constant

Reply:
the mascot for the refactoring guru site is adorable. thanks for sharing the link